File: lib-fast4Android-core/src/main/java/io/github/keep2iron/android/net/interceptor/GetParamsInterceptor.java
package io.github.keep2iron.android.net.interceptor;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * @author keep2iron <a href="http://keep2iron.github.io">Contract me.</a>
 * @version 1.0
 * @since 2017/11/28 16:02
 */
public class GetParamsInterceptor implements Interceptor{
    private Map<String, String> mGetParams;

    public GetParamsInterceptor(Map<String, String> getParams) {
        mGetParams = getParams;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request original = chain.request();
        
        Map<String, String> getParams = new HashMap<>();
        getParams.putAll(mGetParams);
        //添加get参数
        HttpUrl.Builder urlBuilder = original.url().newBuilder();
        for (Map.Entry<String, String> entry : getParams.entrySet()) {
            urlBuilder.addQueryParameter(entry.getKey(), entry.getValue());
        }
        HttpUrl url = urlBuilder.build();
        Request.Builder requestBuilder = original.newBuilder().method(original.method(), original.body()).url(url);
        original = requestBuilder.build();

        Response response = chain.proceed(original);

        return response;
    }
}

File: lib-fast4Android-core/src/main/java/io/github/keep2iron/android/net/interceptor/GetParamsInterceptor.kt
package io.github.keep2iron.android.net.interceptor

import java.io.IOException
import java.util.HashMap

import okhttp3.HttpUrl
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response

/**
 * @author keep2iron [Contract me.](http://keep2iron.github.io)
 * @version 1.0
 * @since 2017/11/28 16:02
 *
 *
 * 统一的GET参数添加的拦截器
 */
class GetParamsInterceptor(val listener: (headerParams: Map<String, String>) -> Unit) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        var original = chain.request()

        val getParams = HashMap<String, String>()
        listener(getParams)
        //添加get参数
        val urlBuilder = original.url().newBuilder()
        for ((key, value) in getParams) {
            urlBuilder.addQueryParameter(key, value)
        }
        val url = urlBuilder.build()
        val requestBuilder = original.newBuilder().method(original.method(), original.body()).url(url)
        original = requestBuilder.build()

        return chain.proceed(original)
    }
}
